plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -275,7 +281,7 @@ private static Block getColumnMetricBlock(RowType columnMetricType, Object min,
         return Partition.toMap(idToTypeMapping, idToMetricMap);
     }
 
-    public static Object convert(Object value, Type type)
+    public static Object convert(Object value, Type type, io.trino.spi.type.Type trinoType)
```

Comment:
The input `type` implies what the `trinoType` is, so passing both looks incorrect.
Otherwise you wouldn't write `if (type instanceof Types.FloatType) {` below without also checking the `trinoType`

let's remove `trinoType` arg from here and let's maybe use 

```
Types.DecimalType icebergDecimalType = (Types.DecimalType) type;
DecimalType trinoDecimalType = createDecimalType(...);
``` 

below